src/NuGet.Core/NuGet.Packaging/ContentModel/ReadOnlyMemoryCharComparerOrdinal.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.ContentModel
+{
+    /// <summary>
+    /// Represents a comparer of <see cref="ReadOnlyMemory{T}" /> that uses string ordinal comparison.
+    /// </summary>
+    internal sealed class ReadOnlyMemoryCharComparerOrdinal : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryCharComparerOrdinal Instance { get; } = new ReadOnlyMemoryCharComparerOrdinal();
+
+        private ReadOnlyMemoryCharComparerOrdinal()
+        {
+        }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+        {
+            return x.Span.Equals(y.Span, StringComparison.Ordinal);
+        }
+
+        public unsafe int GetHashCode(ReadOnlyMemory<char> obj)
```

Comment:
This is crazy complex compared to what you _could_ be doing (as taken from [here](https://dev.azure.com/devdiv/Personal/_git/DotNetApiPortReportProcessor.andarno?path=/BigCsvReader/ReadOnlyMemoryCharComparer.cs)):

```cs
public class ReadOnlyMemoryCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
{
    internal static readonly ReadOnlyMemoryCharComparer Ordinal = new(false);
    internal static readonly ReadOnlyMemoryCharComparer OrdinalIgnoreCase = new(true);

    private readonly CompareOptions options;
    private readonly CompareInfo compareInfo;
    private readonly StringComparer comparer;
    private readonly StringComparison comparison;

    public ReadOnlyMemoryCharComparer(bool ignoreCase)
    {
        this.options = ignoreCase ? CompareOptions.OrdinalIgnoreCase : CompareOptions.Ordinal;
        this.compareInfo = CompareInfo.GetCompareInfo(CultureInfo.CurrentCulture.LCID);
        this.comparer = this.compareInfo.GetStringComparer(this.options);
        this.comparison = ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
    }

    public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y) => x.Span.Equals(y.Span, this.comparison);

    public int GetHashCode(ReadOnlyMemory<char> obj) => this.compareInfo.GetHashCode(obj.Span, this.options);
}
```

Reply:
I guess my suggestion only works if you're targeting .NET. .NET Framework doesn't offer the `CompareInfo.GetHashCode` overload this uses.